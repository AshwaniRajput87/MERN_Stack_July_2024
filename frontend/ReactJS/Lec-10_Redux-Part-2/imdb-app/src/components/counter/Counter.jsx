import { useState } from "react";

import { Button } from "@mui/material";
import { useCounter } from "../../context/counter/useCounter";

const Counter = () => {
  const { count, handleIncrement } = useCounter();
  console.log(count);

  return (
    <>
      <h1>Counter comp</h1>
      <p>Count value: {count}</p>
      <Button onClick={handleIncrement} variant="text">
        +
      </Button>
    </>
  );
};

export default Counter;
