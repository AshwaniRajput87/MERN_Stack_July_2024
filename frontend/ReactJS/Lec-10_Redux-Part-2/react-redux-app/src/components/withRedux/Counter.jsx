import { useDispatch, useSelector } from "react-redux";
import { CounterAction } from "../../redux/slices/CounterSlice";

const Counter = () => {
  const count = useSelector((store) => store.counter.value);

  const dispatch = useDispatch();

  const handleIncrement = () => {
    dispatch(CounterAction.increment());
  };

  // decrement logic do as assignment
  return (
    <>
      <h2>Counter App With Redux</h2>
      <p>Counter value: {count} </p>
      <button onClick={handleIncrement}>+</button>
    </>
  );
};

export default Counter;
