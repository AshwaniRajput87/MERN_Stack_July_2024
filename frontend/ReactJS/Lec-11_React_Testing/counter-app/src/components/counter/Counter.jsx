import React from "react";
import { useState } from "react";
//import "./counter.style.css";

const Counter = () => {
  const [count, setCount] = useState(0);

  const handleIncrement = () => {
    setCount(count + 1);
  };

  const handleDecrement = () => {
    setCount(count - 1);
  };
  return (
    <>
      <div className="counter-container">
        <button onClick={handleDecrement}>-</button>
        <p className="text">{count}</p>
        <button onClick={handleIncrement}>+</button>
      </div>
    </>
  );
};

export default Counter;
