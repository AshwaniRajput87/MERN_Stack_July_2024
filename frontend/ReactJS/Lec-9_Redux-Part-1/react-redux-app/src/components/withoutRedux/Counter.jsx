import { useState } from "react";
const Counter = () => {
  const [count, setCount] = useState(0);
  const handleIncrement = () => {
    setCount(count + 1);
  };
  return (
    <>
      <p>Count Value: {count}</p>

      <button onClick={handleIncrement}>Increment</button>
    </>
  );
};

export default Counter;
