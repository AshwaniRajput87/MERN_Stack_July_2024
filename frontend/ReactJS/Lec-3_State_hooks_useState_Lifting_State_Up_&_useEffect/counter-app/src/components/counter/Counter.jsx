import { useState } from "react";
import "./counter.css";

const Counter = ({ count }) => {
  const [counter, setCounter] = useState(count); // useState() hook always returns an array of two elements, state variable and state update function

  const handleDecrement = () => {
    setCounter(counter - 1);
  };

  return (
    <div className="counter-container">
      <button onClick={handleDecrement}>-</button>
      <p className="count-text">{counter}</p>
      <button
        onClick={() => {
          setCounter(counter + 1);
        }}
      >
        +
      </button>
    </div>
  );
};

export default Counter;
