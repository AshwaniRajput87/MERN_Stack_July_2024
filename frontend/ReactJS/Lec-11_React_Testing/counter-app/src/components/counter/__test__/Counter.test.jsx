import React from "react";
import Counter from "../Counter";

import { render, screen, fireEvent } from "@testing-library/react";

describe("Counter Component", () => {
  test("Sanpshot of counter comp", () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the counter comp with initial value", () => {
    //const {asFragment} = render(<Counter />);
    render(<Counter />);
    const decrementBtn = screen.getByText("-");
    const countText = screen.getByText("0");
    const incrementBtn = screen.getByText("+");

    expect(decrementBtn).toBeTruthy();
    expect(incrementBtn).toBeTruthy();
    expect(countText).toBeTruthy();
  });

  test("increment btn must increase the value of count", () => {
    render(<Counter />);

    const incrementBtn = screen.getByText("+");

    fireEvent.click(incrementBtn);

    const countText = screen.getByText("1");

    expect(countText).toBeTruthy();
  });

  // Assignment: Write the test case for decrement button.
});
