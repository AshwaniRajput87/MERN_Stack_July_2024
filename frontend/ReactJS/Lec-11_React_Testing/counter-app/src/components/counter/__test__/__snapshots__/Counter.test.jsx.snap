// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Sanpshot of counter comp 1`] = `
<DocumentFragment>
  <div
    class="counter-container"
  >
    <button>
      -
    </button>
    <p
      class="text"
    >
      0
    </p>
    <button>
      +
    </button>
  </div>
</DocumentFragment>
`;
